pkg/gds/gds_test.go

```diff
@@ -105,6 +106,160 @@ func (s *gdsTestSuite) TestRegister() {
 	require.Error(err)
 }
 
+// TestLookup test that the Lookup RPC correctly returns details for a VASP.
+func (s *gdsTestSuite) TestLookup() {
+	s.LoadFullFixtures()
+	require := s.Require()
+	ctx := context.Background()
+
+	id := "d9da630e-41aa-11ec-9d29-acde48001122"
+	vasp := s.fixtures[vasps][id].(*pb.VASP)
+
+	// Start the gRPC client
+	require.NoError(s.grpc.Connect())
+	defer s.grpc.Close()
+	client := api.NewTRISADirectoryClient(s.grpc.Conn)
+
+	// Supplied VASP ID does not exist
+	request := &api.LookupRequest{
+		Id: "abc12345-41aa-11ec-9d29-acde48001122",
+	}
+	_, err := client.Lookup(ctx, request)
+	require.Error(err)
+
+	expected := &api.LookupReply{
+		Id:                  id,
+		RegisteredDirectory: vasp.RegisteredDirectory,
+		CommonName:          vasp.CommonName,
+		Endpoint:            vasp.TrisaEndpoint,
+		IdentityCertificate: vasp.IdentityCertificate,
+		Country:             vasp.Entity.CountryOfRegistration,
+		VerifiedOn:          vasp.VerifiedOn,
+		Name:                "CharlieBank",
+	}
+
+	// VASP exists in the database
+	request.Id = id
+	reply, err := client.Lookup(ctx, request)
+	require.NoError(err)
+	require.True(proto.Equal(expected, reply))
+	// TODO: Check that a signing certificate is returned
```

Comment:
I modified the fakerize script to produce signing certificates so I could test this case, but realized that the script is producing new VASP and CertReq IDs every time it's run. This breaks the logic of the tests because we are assuming that the "source of truth" fixture IDs are constant. We might have to modify the tests to not care about specific fixture IDs and use more meaningful names (e.g., "CharlieBank") or modify the fakerize script to produce consistent UUIDs, but that is definitely out of the scope of this PR.

Reply:
Good point! I think either of these suggestions would work -- either fixing the uuids in `fakerize.py` or changing the tests to compare the names rather than uuids. I see that you've opened sc-2282 for us to revisit this, so I think we're good to go here 👍